CARET_trace/src/ros_trace_points.cpp

```diff
@@ -51,6 +51,28 @@
 std::unique_ptr<std::thread> trace_node_thread;
 thread_local bool trace_filter_is_rcl_publish_recorded;
 
+bool exclusion_event = false;
+
+bool filter_launch_node()
+{
+  pid_t pid = getpid();
+  FILE * fp;
+  char cmd_str[1024];
+  char result_str[1024];
+
+  snprintf(cmd_str, sizeof(cmd_str), "ps -ax |grep \"ros2 launch\" | grep %d", pid);
+  fp = popen(cmd_str, "r");
+  if (fgets(result_str, sizeof(result_str), fp) == NULL) {
+    pclose(fp);
+    return false;
+  }
+  pclose(fp);
+  if (strstr(result_str, "grep") == NULL) {
+    return true;
+  }
+  return false;
```

Comment:
- It is better to get command-line from "/proc/self/cmdline". I believe this file contains null-separated command name and arguments.
- It is better to ignore string before tne last "/" of command name to deal with command-lines like "/opt/ros/humble/ros2 launch ..."
- It is better to ignore argments which start with "-" when searching "launch" to deal with command-lines like "ros2 --use-python-default-buffering launch ...")

Reply:
I fixed it with the policy you talked about